Detach input/output streams from main.cpp

wipe2238 opened the pull request:
### What?

Replaced `g_[io]fstream` with `CFalloutScript::m_[io]fstream&` and tweaked related code.

### Why?

While pointless looking on its own, end goal is to make source ready to be built as static library at some point in future.

### Wow!

"Detach" here means replacing hardcore dependency with dependency injection; while int2ssl keeps preparing streams inside `main.cpp`, other application will have it easier to implement different configuration source without it (own command line handling / config file / GUI / smoke signals / etc).

wipe2238 committed:
Detach input/output streams from main.cpp

phobos2077 commented:
Did you test it?

phobos2077 commented on FalloutScript.h:
```diff
@@ -109,6 +110,9 @@ class CFalloutScript
     typedef CMap<uint32_t, uint32_t, CDefObject, CDefObject&> CMapuint32_tToDefObject;
 
+    std::ifstream& m_ifstream;
+    std::ofstream& m_ofstream;
```
If we're abstracting stuff, can't we use istream and ostream instead?

phobos2077 commented on Node.cpp:
```diff
@@ -51,27 +48,27 @@ CNode& CNode::operator = (const CNode& node)
 }
 
-void CNode::StoreTree(int nIndent, int nIndex)
+void CNode::StoreTree(std::ofstream& ofstream, int nIndent, int nIndex)
```
Maybe better to use more "normal" name like outputStream/inputStream (or just `stream`)?

wipe2238 replied:
Sure, i tried to follow current naming (same applies to `m_*stream` mentioned above) but no problem refreshing it a bit. I'll get to it tomorrow morning.

wipe2238 committed:
- I/O streams names more descriptive/human-friendly
- I/O streams wrapped in simple object passed around between various application parts

wipe2238 commented:
> Did you test it?

Yea, everything works like before; tested with both games and a few mods.

---

While renaming, i realized i'm going to need to pass more than just streams in future~

I went ahead and wrapped them inside simple struct, which is passed around during serialization/dump. Should make user-provided options easily accessible even by code deep down in helper classes without need to edit 3/4 of headers for one extra parameter (see `-1` switch needed by `COpcode::GetAttributes(...)`).

phobos2077 commented on Namespace.cpp:
```diff
@@ -27,16 +27,16 @@ CNamespace::~CNamespace()
 }
 
-void CNamespace::Serialize(std::ifstream& ifstream)
+void CNamespace::Serialize(CFalloutScriptData& scriptData)
```
You should only pass the thing you're using to the function. Otherwise such interface implies function can read and write, which isn't the case.
